### PR TITLE
feat(observability): add node pool creation SLO dashboard

### PR DESCRIPTION
**Pull request by sravyanimmala**
https://redhat.atlassian.net/browse/ARO-26186

## What

Adds a new Grafana dashboard for the Node Pool Creation user journey, covering the ADR's 5 baseline SLIs:

Availability: current % with SLO target line
Errors: error rate over time with SLO threshold
Latency: p50/p95/p99 create duration with SLO target line
Traffic: create attempt rate with trend
Saturation: backend controller pod CPU/memory/replicas/restarts

Also includes SLO breach visibility (color thresholds, error budget panel) 
and a "Node Pools Deleted" panel for fleet-count context.

## Why

Per the ARO HCP Alerting Recommendation ADR, every user journey needs a dashboard with baseline SLI coverage and SLO breach visibility.

## Testing

Checked every panel against real fleet data on arohcp-dev

special notes for your reviewer:
SLO targets (99% success, 15-min p95) come from the comment on ARO-19992, not finalized numbers. [ARO-26185](https://redhat.atlassian.net/browse/ARO-26185) may need to update these once that lands.

## PR Checklist

[]PR is scoped to a single task (no mixed concerns)
[x]Title follows Conventional Commits format
[x]Summary explains the "why" behind the change
[x] Linked to relevant ticket/issue
[x] Screenshots included - 
[nodepool slo dashboard.pdf](https://github.com/user-attachments/files/32130135/nodepool.slo.dashboard.pdf)

[x] Self-reviewed the diff
[x] CI/CD checks are passing (ignore Tide)
[x] Commit history is clean (rebased/squashed)
[x] Specific reviewers tagged
[x] All comment threads resolved before merge

### CODE REVIEW

**Comment by komidore64** [comment]
@sravyanimmala do you think this should be an entirely new dashboard, or incorporate any new metrics/queries into the existing "Node Pool Management SLO Dashboard"?

Please let me know your thoughts.

**Comment by sravyanimmala** [comment]
> Lakshmi Sravya Nimmala (@sravyanimmala) do you think this should be an entirely new dashboard, or incorporate any new metrics/queries into the existing "Node Pool Management SLO Dashboard"?
> 
> Please let me know your thoughts.

I kept it separate for the provisoning` SLO- latency, error budget and the slo targets only make sense for the  node pool creation. Happy to discuss if you'd rather combine them

**Comment by komidore64** [comment]
> I kept it separate for the provisoning` SLO- latency, error budget and the slo targets only make sense for the node pool creation. Happy to discuss if you'd rather combine them

On second thought, that makes sense. They only make sense for creation, but once creation is done they're no longer relevant.

**Comment by komidore64** [comment]
Suggestion: Please mention in this PR's title that it's for node pool _creation_, as we already have a Node Pool Management SLO dashboard.

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json [hunk 1]
```diff
@@ -0,0 +1,2719 @@
...
+          "expr": "(quantile(0.50, ((max without (prometheus_replica) (backend_resource_operation_last_transition_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=\"succeeded\", subscription_id!~\"${exclude_subs:raw}\"}) - max without (prometheus_replica) (backend_resource_operation_start_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=\"succeeded\", subscription_id!~\"${exclude_subs:raw}\"})) / 60) and ((time() - max without (prometheus_replica) (backend_resource_operation_last_transition_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=\"succeeded\", subscription_id!~\"${exclude_subs:raw}\"})) < 3600)) by (cluster)) or (count by (cluster) (max without (prometheus_replica) (backend_resource_operation_phase_info{resource_type=\"$resource_type\", operation_type=\"create\", phase=\"succeeded\", subscription_id!~\"${exclude_subs:raw}\"}) and (time() - max without (prometheus_replica) (backend_resource_operation_last_transition_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=\"succeeded\", subscription_id!~\"${exclude_subs:raw}\"}) < 3600)) * 0)",
```
Comment (swiencki) [inline observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json]: On the flagged concerns here: this `quantile(0.5, X) by (cluster)` target is valid PromQL and every panel already removes the HA `prometheus_replica` dimension, but the availability numerator's `< 900s` gate that blends latency into availability, counting `canceled` as an error, and the gauge-based 7d/30d error-budget windows are all worth addressing.

Comment (sravyanimmala) [reply observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json]: Thanks . Addressed all three. Split timeliness into its own panel (Availability is now plain success/failure). Canceled no longer counts as an error in the SLO calcs, just tracked separately in the Failed/Canceled panels. Renamed 30d → 7d and fixed the window to match the real Cosmos TTL.

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json [hunk 2]
```diff
@@ -0,0 +1,2719 @@
...
+        "options": [],
+        "query": "prometheus",
+        "refresh": 1,
+        "regex": "^Managed_Prometheus_services-.*$",
+        "type": "datasource"
+      },
+      {
+        "current": {
+          "text": "",
+          "value": ""
```
[the remark this answers is not in the document]
Comment (sravyanimmala) [reply observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json:2824]: Empty regex in Prometheus only matches the empty string, so subscription_id!~"${exclude_subs:raw}" actually matches everything and excludes nothing — that's the correct default, not a bug and also seen real data every time we tested with this box empty. No change needed.

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json [hunk 3]
```diff
@@ -0,0 +1,2719 @@
...
+        },
+        {
+          "id": "calculateField",
+          "options": {
+            "alias": "Points Above Target",
+            "binary": {
+              "left": "Availability %",
+              "operator": "-",
+              "reducer": "sum",
+              "right": "99"
```
[the remark this answers is not in the document]
Comment (sravyanimmala) [reply observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json:1022]: Tested and working. This panel has shown correct, real percentages every time we checked it live. If "99" were treated as an undefined field, it'd show "No data" every time, and it never has. No change needed.

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sravyanimmala) [reply observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json]: No change needed.Verified with the actual Prometheus parser library — both quantile(0.5, X) by (cluster) and quantile by (cluster) (0.5, X) are valid.

**Review by komidore64 — APPROVED** [review]

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json [hunk 4]
```diff
@@ -0,0 +1,2868 @@
...
+        "type": "constant"
+      }
+    ]
+  },
+  "time": {
+    "from": "now-24h",
+    "to": "now"
+  },
+  "timepicker": {},
+  "timezone": "browser",
```
Comment (swiencki) [inline observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json]: Set `timezone` to `utc` instead of `browser` so fleet burn windows and terminal-transition timestamps read identically for every operator regardless of local time zone.

File: observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json [hunk 5]
```diff
@@ -0,0 +1,2868 @@
...
...subscription_id, resource_id, resource_type, operation_type, phase) (backend_resource_operation_phase_info{resource_type=\"$resource_type\", operation_type=\"create\", phase=~\"succeeded|failed\", subscription_id!~\"${exclude_subs:raw}\"}) and (time() - max by (cluster, environment, region, subscription_id, resource_id, resource_type, operation_type, phase) (backend_resource_operation_last_transition_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=~\"succeeded|failed\", subscription_id!~\"${exclude_subs:raw}\"}) < 3600)) * 0))\n  /\n  count by (cluster) (max by (cluster, environment, region, subscription_id, resource_id, resource_type, operation_type, phase) (backend_resource_operation_phase_info{resource_type=\"$resource_type\", operation_type=\"create\", phase=~\"succeeded|failed\", subscription_id!~\"${exclude_subs:raw}\"}) and (time() - max by (cluster, environment, region, subscription_id, resource_id, resource_type, operation_type, phase) (backend_resource_operation_last_transition_time_seconds{resource_type=\"$resource_type\", operation_type=\"create\", phase=~\"succeeded|failed\", subscription_id!~\"${exclude_subs:raw}\"}) < 3600))\n) * 100",
```
Comment (swiencki) [inline observability/grafana-dashboards/sre/user-journey/nodepool-creation-slo.json]: Once the node pool creation SLI/SLO work in ARO-26179 lands, consider sourcing these availability panels from a shared recording rule and adding the matching alert, so the dashboard and the alert enforce one definition of the 99% availability and 15 minute timeliness target.
